src/cloud_provider/aws/databases/utilities.rs

```diff
@@ -74,3 +83,82 @@ pub fn delete_terraform_tfstate_secret(
         }
     }
 }
+
+pub fn get_version_semver(version: &str) -> Result<VersionsNumber, StringError> {
+    let mut version_split = version.split(".");
+
+    let major = match version_split.next() {
+        Some(major) => major.to_string(),
+        _ => {
+            return Err(StringError::new(
+                "please check the version you've sent, it can't be checked".to_string(),
+            ))
+        }
+    };
+
+    let minor = match version_split.next() {
+        Some(minor) => Some(minor.to_string()),
+        _ => None,
+    };
+
+    let patch = match version_split.next() {
+        Some(patch) => Some(patch.to_string()),
+        _ => None,
+    };
+
+    Ok(VersionsNumber {
+        major,
+        minor,
+        patch,
+    })
+}
+
+pub fn get_supported_version_to_use(
+    all_supported_versions: HashMap<&str, &str>,
```

Comment:
I'd suggest using type alias for hashmap key and value, then we can understand what types are expected?

Reply:
@deimosfr ->

```rust
type ProvidedVersion<'a> = &'a str;
type RealVersion<'a> = &'a str;

//....
pub fn get_supported_version_to_use(
all_supported_versions: HashMap<ProvidedVersion<'a>, RealVersion<'a>>,...
)
```